atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/SweepCommand.java

```diff
@@ -0,0 +1,170 @@
+ */
+package com.palantir.atlasdb.cli;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.google.common.base.Functions;
+import com.google.common.base.Optional;
+import com.google.common.base.Predicates;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.common.io.BaseEncoding;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.cli.api.AtlasDbServices;
+import com.palantir.atlasdb.cli.api.SingleBackendCommand;
+import com.palantir.atlasdb.config.AtlasDbConfig;
+import com.palantir.atlasdb.keyvalue.api.SweepResults;
+import com.palantir.atlasdb.schema.generated.SweepPriorityTable;
+import com.palantir.atlasdb.schema.generated.SweepTableFactory;
+import com.palantir.atlasdb.sweep.SweepTaskRunner;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.atlasdb.transaction.impl.TxTask;
+import com.palantir.common.base.Throwables;
+
+import io.airlift.airline.Command;
+import io.airlift.airline.Option;
+
+@Command(name = "sweep", description = "Sweep old table rows")
+public class SweepCommand extends SingleBackendCommand {
+
+    @Option(name = {"-n", "--namespace"},
+            description = "An atlas namespace to sweep")
+    String namespace;
+
+    @Option(name = {"-t", "--table"},
+            description = "An atlas table to sweep")
+    String table;
+
+    @Option(name = {"-r", "--row"},
+            description = "A row to start from (hex encoded bytes)")
+    String row;
+
+    @Option(name = {"-a", "--all"},
+            description = "Sweep all tables")
+    boolean sweepAllTables;
+
+	@Override
+	protected int execute(final AtlasDbServices services) {
+        AtlasDbConfig config = services.getServerConfig();
+        SweepTaskRunner sweepRunner = services.getSweepTaskRunner();
+
+        if (!((namespace != null) ^ (table != null) ^ sweepAllTables)) {
+            System.err.println("Specify one of --namespace, --table, or --all options.");
+            return 1;
+        }
+        if ((namespace != null) && (row != null)) {
+            System.err.println("Cannot specify a start row (" + row + ") when sweeping multiple tables (in namespace " + namespace + ")");
+            return 1;
+        }
+
+        Map<String, Optional<byte[]>> tableToStartRow = Maps.newHashMap();
+
+        if ((table != null)) {
+            Optional<byte[]> startRow = Optional.of(new byte[0]);
+            if (row != null) {
+                startRow = Optional.of(decodeStartRow(row));
+            }
+            tableToStartRow.put(table, startRow);
+        } else if (namespace != null) {
+            Set<String> tablesInNamespace = Sets.filter(services.getKeyValueService().getAllTableNames(),
+                    Predicates.containsPattern("^" + namespace + "\\."));
```

Comment:
This is how we do it in pg.  We should change the api in general to be namespace aware (pass TableReference all the way down instead of having untyped strings), but that is a much bigger change.